src/Microsoft.Identity.Client/Internal/OAuth2/OAuth2Client.cs

```diff
@@ -88,18 +88,34 @@ public void AddBodyParameter(string key, string value)
             }
 
             HttpResponse response = null;
-
-            if (method == HttpMethod.Post)
+            Uri endpointUri = CreateFullEndpointUri(endPoint);
+            var httpEvent = new HttpEvent()
+            {
+                HttpPath = EventBase.ScrubTenant(endpointUri),
```

Comment:
This is not the intent in the design. the code to scrub should be in the setter, not in the main class.